src/autopas/containers/octree/Octree.h

```diff
@@ -0,0 +1,240 @@
+         const double cellSizeFactor)
+      : CellBasedParticleContainer<ParticleCell>(boxMin, boxMax, cutoff, skin) {
+    // TODO(johannes): Obtain this from a configuration, reported in https://github.com/AutoPas/AutoPas/issues/624
+    int unsigned treeSplitThreshold = 16;
+
+    double interactionLength = this->getInteractionLength();
+    this->_cells.push_back(
+        OctreeNodeWrapper<Particle>(boxMin, boxMax, treeSplitThreshold, interactionLength, cellSizeFactor));
+
+    // Extend the halo region with cutoff + skin in all dimensions
+    auto haloBoxMin = utils::ArrayMath::subScalar(boxMin, this->getInteractionLength());
+    auto haloBoxMax = utils::ArrayMath::addScalar(boxMax, this->getInteractionLength());
+    this->_cells.push_back(
+        OctreeNodeWrapper<Particle>(haloBoxMin, haloBoxMax, treeSplitThreshold, interactionLength, cellSizeFactor));
+  }
+
+  [[nodiscard]] std::vector<ParticleType> updateContainer() override {
+    // This is a very primitive and inefficient way to recreate the container:
+    // 1. Copy all particles out of the container
+    // 2. Clear the container
+    // 3. Insert the particles back into the container
+
+    // leaving: all outside boxMin/Max
+
+    // TODO(johannes): Make this less indirect. (Find a better way to iterate all particles inside the octree to change
+    //  this function back to a function that actually copies all particles out of the octree.)
+    //  The problem is captured by https://github.com/AutoPas/AutoPas/issues/622
+    std::vector<Particle *> particleRefs;
+    this->_cells[CellTypes::OWNED].appendAllParticles(particleRefs);
+    std::vector<Particle> particles;
+    auto result = std::vector<ParticleType>();
+    for (auto *p : particleRefs) {
+      if (utils::inBox(p->getR(), this->getBoxMin(), this->getBoxMax())) {
+        particles.push_back(*p);
+      } else {
+        result.push_back(*p);
+      }
+    }
+    this->deleteAllParticles();
+
+    for (auto &particle : particles) {
+      addParticleImpl(particle);
+    }
+
+    // This should happen according to the documentation in the ParticleContainerInterface#updateContainer
+    deleteHaloParticles();
+
+    return result;
+  }
+
+  void iteratePairwise(TraversalInterface *traversal) override {
+    auto *traversalInterface = dynamic_cast<OTTraversalInterface<ParticleCell> *>(traversal);
+    if (traversalInterface) {
+      traversalInterface->setCells(&this->_cells);
+    }
+
+    traversal->initTraversal();
+    traversal->traverseParticlePairs();
+    traversal->endTraversal();
+  }
+
+  /**
+   * @copydoc ParticleContainerInterface::getContainerType()
+   */
+  [[nodiscard]] ContainerOption getContainerType() const override { return ContainerOption::octree; }
+
+  /**
+   * @copydoc ParticleContainerInterface::getParticleCellTypeEnum()
+   */
+  [[nodiscard]] CellType getParticleCellTypeEnum() override { return CellType::FullParticleCell; }
```

Comment:
I think it might be worth mentioning that this container does not actually use the FullParticleCell but something that is sufficiently similar.

Reply:
Do you mean that I should add a comment that says that or should I add another custom `CellType`?

Reply:
just expand the documentation.

Reply:
In 90c931d I added a note to the documentation: 

https://github.com/AutoPas/AutoPas/blob/90c931d1df57dbc85912c3d37cc09ba1055a4016/src/autopas/containers/octree/Octree.h#L129-L135

Let me know whether that's ok for you :)